### PR TITLE
Add file provider

### PR DESCRIPTION
**Pull request by ygorbarboza**
Solve `android.os.FileUriExposedException: file.jpg exposed beyond app through ClipData.Item.getUri()` exception on targetSdk 24+

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-share",
   "description": "Social Share, Sending Simple Data to Other Apps",
-  "version": "1.0.23",
+  "version": "999.0.28",
```
Comment (jgcmarins) [inline package.json]: why is that?

**Comment by ygorbarboza** [comment]
@jgcmarins sorry about that. Actually this PR was not suppose to be created now. This change is only relevante when React Native uses a new targetSdkVersion. For now there is no need for this

**Comment by deliveryblueprints** [comment]
Thanks for this, works great. One thing to note to get this working in your project is to implement the ShareApplication.java in your MainApplication.java.
Then override the getFileProviderAuthority like so..
`  @Override
    public String getFileProviderAuthority() {
        return BuildConfig.APPLICATION_ID + ".fileprovider";
    }
`

This will properly populate the xml with your applicationId, i.e. com.myapporg.myappname.fileprovider

Thanks again!

**Comment by ER1011** [comment]
@deliveryblueprints, what exactly did you put in the MainApplication.java and where does the @Override part go? I've tried in multiple ways but can't get it to work. Thank you

**Comment by ygorbarboza** [comment]
@ER1011 Just do as said here, make your application extends `ShareApplication` and create function`getFileProviderAuthority` with the same name as the provider-> android:authorities on Manifest

**Comment by ER1011** [comment]
@ygorbarboza the link you posted seems to be broken

**Comment by ygorbarboza** [comment]
@ER1011 sorry about that
https://developer.android.com/training/secure-file-sharing/setup-sharing.html

**Comment by jgcmarins** [comment]
probably solves https://github.com/EstebanFuentealba/react-native-share/issues/185

**Comment by ygorbarboza** [comment]
@jgcmarins Yup

**Comment by ygorbarboza** [comment]
This can be merged, the problem is that we gonna have to bump the compileSdkVersion and the buildTools

**Comment by jgcmarins** [comment]
@EstebanFuentealba and @sibelius can we merge this PR?

**Comment by bitshadow** [comment]
Thanks @ygorbarboza for PR. We really needed this in our app.

**Comment by bitshadow** [comment]
@sibelius Looks like this is working for us can you release the new package in NPM if possible. Thanks.

**Pull request merged**